apps/desktop/src/lib/project/project.ts

```diff
@@ -162,10 +162,32 @@ export function handleAddProjectOutcome(
 			return true;
 		case 'notAGitRepository':
 			showToast({
-				testId: TestId.AddProjectNotAGitRepoModal,
-				style: 'warning',
 				title: 'Not a Git repository',
-				message: `Unable to add project: ${outcome.subject}`
+				message:
+					'The selected directory is not a Git repository. Would you like to initialize one?',
+				style: 'warning',
+				extraAction: {
+					label: 'Initialize Repository',
+					onClick: async (dismiss) => {
+						try {
```

Comment:
Here is the main difficulty: initialising and re-adding the project from here to get a project id.